provider/resources.go

```diff
@@ -2435,8 +2407,20 @@ func Provider() tfbridge.ProviderInfo {
 			"aws_shield_protection_health_check_association": {Tok: awsResource(shieldMod, "ProtectionHealthCheckAssociation")},
 
 			// S3
-			"aws_s3_account_public_access_block": {Tok: awsResource(s3Mod, "AccountPublicAccessBlock")},
-			"aws_s3_bucket": {
+			"aws_s3_account_public_access_block":                 {Tok: awsResource(s3Mod, "AccountPublicAccessBlock")},
+			"aws_s3_bucket":                                      {Tok: awsResource(s3Mod, "BucketV2")},
```

Comment:
We should add a comment here explaining the scope of our decision, why, and when we can un-alias (v6.0), plus a link to the design doc.